test/solidity/Facets/AllBridgeFacet.t.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.17;
+
+import { LibAllowList, TestBaseFacet, console, ERC20 } from "../utils/TestBaseFacet.sol";
+import { AllBridgeFacet, IAllBridge, MessengerProtocol } from "lifi/Facets/AllBridgeFacet.sol";
+import { TestBase } from "../utils/TestBase.sol";
+
+// Stub AllBridgeFacet Contract
+contract TestAllBridgeFacet is AllBridgeFacet {
+    constructor(IAllBridge _allBridge) AllBridgeFacet(_allBridge) {}
+
+    function addDex(address _dex) external {
+        LibAllowList.addAllowedContract(_dex);
+    }
+
+    function setFunctionApprovalBySignature(bytes4 _signature) external {
+        LibAllowList.addAllowedSelector(_signature);
+    }
+}
+
+contract AllBridgeFacetTest is TestBaseFacet {
+    // These values are for Optimism_Kovan
+    address internal constant BRIDGE_ADDRESS =
+        0xA314330482f325D38A83B492EF6B006224a3bea9;
+    address internal constant ALLBRIDGE_POOL =
+        0x1D3df13aDAe6cA91Fb90b977c21d6e90ad8d403C;
+    // -----
+    AllBridgeFacet.AllBridgeData internal validAllBridgeData;
+    TestAllBridgeFacet internal allBridgeFacet;
+
+    function setUp() public {
+        customBlockNumberForForking = 16661275;
+        initTestBase();
+
+        allBridgeFacet = new TestAllBridgeFacet(IAllBridge(BRIDGE_ADDRESS));
+        bytes4[] memory functionSelectors = new bytes4[](4);
+        functionSelectors[0] = allBridgeFacet
+            .startBridgeTokensViaAllBridge
+            .selector;
+        functionSelectors[1] = allBridgeFacet
+            .swapAndStartBridgeTokensViaAllBridge
+            .selector;
+        functionSelectors[2] = allBridgeFacet.addDex.selector;
+        functionSelectors[3] = allBridgeFacet
+            .setFunctionApprovalBySignature
+            .selector;
+
+        addFacet(diamond, address(allBridgeFacet), functionSelectors);
+        allBridgeFacet = TestAllBridgeFacet(address(diamond));
+        allBridgeFacet.addDex(ADDRESS_UNISWAP);
+        allBridgeFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForTokens.selector
+        );
+        allBridgeFacet.setFunctionApprovalBySignature(
+            uniswap.swapTokensForExactETH.selector
+        );
+        allBridgeFacet.setFunctionApprovalBySignature(
+            uniswap.swapETHForExactTokens.selector
+        );
+
+        setFacetAddressInTestBase(address(allBridgeFacet), "AllBridgeFacet");
+
+        // adjust bridgeData
+        bridgeData.bridge = "allbridge";
+        bridgeData.destinationChainId = 137;
+
+        uint256 testPolygonMessengerFees = 69857330070000;
+        uint256 testPolygonBridgeFees = 28764782970000;
+        uint256 fees = testPolygonBridgeFees + testPolygonMessengerFees;
+        // produce valid AllBridgeData
+        validAllBridgeData = AllBridgeFacet.AllBridgeData({
+            fees: fees,
+            recipient: 0x00000000000000000000000012561cc3ea2a60aa158b0421010859a983bf3c96,
+            destinationChainId: 5,
```

Comment:
Why does this differ from the destinationChainId in the bridgeData?

Reply:
It's of type uint8. Is it cheaper to cast the `LiFi.BridgeData.DestinationChainId` to a uint8 than passing an extra field in the struct?